src/System.Net.HttpListener/tests/HttpListenerResponseTests.Headers.cs

```diff
@@ -808,5 +852,29 @@ public static IEnumerable<object[]> ProtocolVersion_Set_TestData()
                 Assert.Equal(0, response.Headers.Count);
             }
         }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotOneCoreUAP))]
+        [ActiveIssue(20425, TargetFrameworkMonikers.Netcoreapp)]
+        public async Task Headers_SetLongName_ThrowsArgumentOutOfRangeException()
+        {
+            HttpListenerResponse response = await GetResponse();
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("value", () => response.Headers["name"] = s_longString);
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("value", () => response.Headers.Set("name", s_longString));
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("value", () => response.Headers.Add("name", s_longString));
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("value", () => response.Headers.Add($"name:{s_longString}"));
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("value", () => response.Headers.Add(HttpResponseHeader.Age, s_longString));
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotOneCoreUAP))]
+        [ActiveIssue(20425, TargetFrameworkMonikers.Netcoreapp)]
```

Comment:
This issue is about long header values.  Is it the right issue?  Aren't these really just another manifestation of the existing issue https://github.com/dotnet/corefx/issues/18784?